Reword matrix multiplier example docs (#2453)
cocotb/cocotb: documentation/source/examples.rst

and ``c_o`` is calculated,
``valid_o`` goes high to signal a valid output value.

A reusable ``DataValidMonitor`` class is defined.
It monitors a streaming data/valid bus,
samples the bus when a transaction occurs,
and places those transactions into an asynchronous :class:`~cocotb.queue.Queue`.

> The queue allows another coroutine to consume monitored transactions at it's own pace.

Reviewer: ```suggestion
The queue allows another coroutine to consume monitored transactions at its own pace.
```

A reusable ``MatrixMultiplierTester`` is also defined.
It instantiates two of the ``DataValidMonitor``\ s:
one to monitor the matrix multiplier input,
and another to monitor the output.

> The ``MatrixMultiplierTester`` :func:`~cocotb.fork`\ s a coroutine which consumes transaction from the input monitor,

Reviewer: ```suggestion
The ``MatrixMultiplierTester`` :func:`~cocotb.fork`\ s a coroutine which consumes transactions from the input monitor,
```

> The ``MatrixMultiplierTester`` then creates a :class:`coroutine <cocotb.fork>` which consumes transaction from the input monitor,

Reviewer 2: ```suggestion
The ``MatrixMultiplierTester`` :func:`~cocotb.fork`\ s a coroutine which consumes transaction from the input monitor,
```
since `fork` is not a `class`

> Then it creates a :class:`coroutine <cocotb.fork>` which consumes transaction from the
> input monitor, feeds them into a model to compute an expected output,

Author: ```suggestion
The ``MatrixMultiplierTester`` then creates a :class:`coroutine <cocotb.fork>` which consumes transaction from the input monitor,
feeds them into a model to compute an expected output,
```

> and finally compares the two for correctness.

Reviewer 3: ```suggestion
and finally compares the module output to the expected for correctness.
```
